<robot name="FordCar">
    <!-- <link name="map"></link> -->

    <link name="body">
        <!-- <inertial>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <mass value="8000"/>
            <inertia ixx="6093" ixy="0" ixz="0" iyy="10567" iyz="0" izz="12807" />
        </inertial> -->

        <visual>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <box size="3.6 2.5 1.7" />
            </geometry>
            <material name="Blue">
                <color rgba="0 0 1.0 1.0"/>
            </material>
        </visual>

        <!-- <collision>
            <origin xyz="1.8 1.25 0.85" rpy="0 0 0"/>
            <geometry>
                <box size="3.6 2.5 1.7"/>
            </geometry>
        </collision> -->
    </link>

    <link name="left_back_wheel">
        <!-- <inertial>
            <origin xyz="0 0 0" rpy="1.57 0 0"/>
            <mass value="10"/>
            <inertia ixx="0.283" ixy="0" ixz="0" iyy="0.283" iyz="0" izz="0.4993" />
        </inertial> -->

        <visual>
            <origin xyz="0 0 0" rpy="1.57 0 0" />
            <geometry>
                <cylinder radius="0.316" length="0.2"/>
            </geometry>
            <material name="Black">
                <color rgba="0 0 0 1.0"/>
            </material>
        </visual>
<!-- 
        <collision>
            <origin xyz="0 0 0" rpy="1.57 0 0"/>
            <geometry>
                <cylinder radius="0.316" length="0.2"/>
            </geometry>
        </collision> -->
    </link>

    <link name="right_back_wheel">
        <!-- <inertial>
            <origin xyz="0 0 0" rpy="1.57 0 0"/>
            <mass value="10"/>
            <inertia ixx="0.283" ixy="0" ixz="0" iyy="0.283" iyz="0" izz="0.4993" />
        </inertial> -->

        <visual>
            <origin xyz="0 0 0" rpy="1.57 0 0" />
            <geometry>
                <cylinder radius="0.316" length="0.2"/>
            </geometry>
            <material name="Black">
                <color rgba="0 0 0 1.0"/>
            </material>
        </visual>
<!-- 
        <collision>
            <origin xyz="0 0 0" rpy="1.57 0 0"/>
            <geometry>
                <cylinder radius="0.316" length="0.2"/>
            </geometry>
        </collision> -->
    </link>

    <link name="front_left_wheel">
        <!-- <inertial>
            <origin xyz="0 0 0" rpy="1.57 0 0"/>
            <mass value="10"/>
            <inertia ixx="0.283" ixy="0" ixz="0" iyy="0.283" iyz="0" izz="0.4993" />
        </inertial> -->

        <visual>
            <origin xyz="0 0 0" rpy="1.57 0 0" />
            <geometry>
                <cylinder radius="0.316" length="0.2"/>
            </geometry>
            <material name="Black">
                <color rgba="0 0 0 1.0"/>
            </material>
        </visual>
<!-- 
        <collision>
            <origin xyz="0 0 0" rpy="1.57 0 0"/>
            <geometry>
                <cylinder radius="0.316" length="0.2"/>
            </geometry>
        </collision> -->
    </link>

    <link name="front_right_wheel">
        <!-- <inertial>
            <origin xyz="0 0 0" rpy="1.57 0 0"/>
            <mass value="10"/>
            <inertia ixx="0.283" ixy="0" ixz="0" iyy="0.283" iyz="0" izz="0.4993" />
        </inertial> -->

        <visual>
            <origin xyz="0 0 0" rpy="1.57 0 0" />
            <geometry>
                <cylinder radius="0.316" length="0.2"/>
            </geometry>
            <material name="Black">
                <color rgba="0 0 0 1.0"/>
            </material>
        </visual>
<!-- 
        <collision>
            <origin xyz="0 0 0" rpy="1.57 0 0"/>
            <geometry>
                <cylinder radius="0.316" length="0.2"/>
            </geometry>
        </collision> -->
    </link>

    <link name="front_middle_wheel">
        <!-- <inertial>
            <origin xyz="0 0 0" rpy="1.57 0 0"/>
            <mass value="10"/>
            <inertia ixx="0.283" ixy="0" ixz="0" iyy="0.283" iyz="0" izz="0.4993" />
        </inertial>

        <visual>
            <origin xyz="0 0 0" rpy="1.57 0 0" />
            <geometry>
                <cylinder radius="0.316" length="0.2"/>
            </geometry>
            <material name="Black">
                <color rgba="0 0 0 1.0"/>
            </material>
        </visual> -->
<!-- 
        <collision>
            <origin xyz="0 0 0" rpy="1.57 0 0"/>
            <geometry>
                <cylinder radius="0.316" length="0.2"/>
            </geometry>
        </collision> -->
    </link>

    <!-- <joint name="map_body" type="floating">
        <origin xyz="0 0 0.85" rpy="0 0 0"/>
        <parent link="map"/>
        <child link="body"/>
    </joint> -->

    <joint name="left_back_wheel_joint" type="fixed">
        <origin xyz="-1.8 1.25 -0.85" rpy="0 0 0"/>
        <parent link="body"/>
        <child link="left_back_wheel"/>
    </joint>

    <joint name="right_back_wheel_joint" type="fixed">
        <origin xyz="-1.8 -1.25 -0.85" rpy="0 0 0"/>
        <parent link="body"/>
        <child link="right_back_wheel"/>
    </joint>

    <joint name="front_middle_wheel_joint" type="revolute">
        <origin xyz="1.8 0 -0.85" rpy="0 0 0"/>
        <parent link="body"/>
        <child link="front_middle_wheel"/>
        <axis xyz="0 0 1"/>
        <limit effort="5" velocity="0.157" lower="-0.785398" upper="0.785398" />
    </joint>

    <joint name="front_middle_front_left_wheel_joint" type="revolute">
        <origin xyz="0 1.25 0" rpy="0 0 0"/>
        <parent link="front_middle_wheel"/>
        <child link="front_left_wheel"/>
        <axis xyz="0 0 1"/>
        <limit effort="5" velocity="0.157" lower="-0.2065" upper="0.2065" />
    </joint>

    <joint name="front_middle_front_right_wheel_joint" type="revolute">
        <origin xyz="0 -1.25 0" rpy="0 0 0"/>
        <parent link="front_middle_wheel"/>
        <child link="front_right_wheel"/>
        <axis xyz="0 0 1"/>
        <limit effort="5" velocity="0.157" lower="-0.2065" upper="0.2065" />
    </joint>

</robot>